src/Microsoft.AspNet.Mvc.TagHelpers/FormTagHelper.cs

```diff
@@ -124,6 +131,14 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             }
 
             var antiforgeryDefault = true;
+            if (Method != null)
+            {
+                output.CopyHtmlAttribute(nameof(Method), context);
+                if (string.Equals(Method, FormMethod.Get.ToString(), StringComparison.OrdinalIgnoreCase))
+                {
+                    antiforgeryDefault = false;
+                }
```

Comment:
Move this inner block down into the `else` at [line 163)(https://github.com/aspnet/Mvc/pull/3933/files#diff-8bd6f4dfc2f4a07e5b08261979376682R163). Note you don't need to double-check `Method != null` there.